### PR TITLE
CMR-6318 updated re-indexing to work better with es7

### PR DESCRIPTION
**Pull request by jaybarra**
Fixed re-indexing tests to properly test re-indexing code

### CODE REVIEW

File: indexer-app/src/cmr/indexer/services/index_service.clj [hunk 1]
```diff
@@ -267,25 +280,28 @@
...
+                             flatten
+                             (remove anti-value-suggestion?))]
+    suggestion-docs))
+
 
 (defn- reindex-suggestions-for-provider
   "Reindex autocomplete suggestion for a given provider"
   [context provider-id]
-  (info "Reindexing latest autocomplete suggestions for provider" provider-id)
+  (info "Reindexing autocomplete suggestions for provider " provider-id)
```
Comment (ygliuvt) [inline indexer-app/src/cmr/indexer/services/index_service.clj:294]: no need to add the space here. Logger will automatically do that.

File: indexer-app/src/cmr/indexer/services/index_service.clj [hunk 2]
```diff
@@ -267,25 +280,28 @@
...
   (let [latest-collection-batches (meta-db/find-in-batches
-                                    context
-                                    :collection
-                                    REINDEX_BATCH_SIZE
-                                    {:provider-id provider-id :latest true})
+                                   context
```
Comment (ygliuvt) [inline indexer-app/src/cmr/indexer/services/index_service.clj:296]: The indentation is off. The original indentation is correct. See https://wiki.earthdata.nasa.gov/display/CMR/Coding+Style+Guidelines

File: indexer-app/src/cmr/indexer/services/index_service.clj [hunk 3]
```diff
@@ -328,10 +344,10 @@
      (when (or (nil? all-revisions-index?) (not all-revisions-index?))
        (info "Reindexing latest collections for provider" provider-id)
        (let [latest-collection-batches (meta-db/find-in-batches
-                                         context
-                                         :collection
-                                         REINDEX_BATCH_SIZE
-                                         {:provider-id provider-id :latest true})]
+                                        context
```
Comment (ygliuvt) [inline indexer-app/src/cmr/indexer/services/index_service.clj:347]: Indentations are off. See above comment for link to coding standard. This applies to lots of the lines in the PR. Seems like you autoformatter is not set up properly.

File: system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj [hunk 4]
```diff
@@ -105,29 +103,25 @@
...
-      {:score 0.20964509, :type "organization", :value "ACRIM SCF", :fields "ACRIM SCF"}
-      {:score 0.16771607, :type "organization", :value "Langley DAAC User Services", :fields "Langley DAAC User Services"}]
-
-     "more complete match"
-     "q=solar irradiation"
-     [{:score 5.0869484, :type "science_keywords", :value "Solar Irradiance", :fields "Sun-Earth Interactions:Solar Activity:Solar Irradiance"}
-      {:score 2.7943783, :type "science_keywords", :value "Solar Irradiance", :fields "Atmosphere:Atmospheric Radiation:Solar Irradiance"}
-      {:score 0.061936468, :type "organization", :value "ACRIM SCF", :fields "ACRIM SCF"}
-      {:score 0.049549174, :type "organization", :value "Langley DAAC User Services", :fields "Langley DAAC User Services"}])))
+  (testing "verify no results come back before reindexing has occurred"
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj:106]: I know you didn't do this, but can you move the collection ingest part from the fixture setup down to the test. That was not done according to our coding standard. The fixture should only have the provider and permission setup in general.

File: system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj [hunk 5]
```diff
@@ -105,29 +103,25 @@
...
+    (let [before (get-in [:feed :entry] (search/get-autocomplete-json "q=sol"))]
+      ;; Verify index does not return results before re-indexing
+      (is (= 0 (count before)))))
+  
+  (testing "after running a reindex values should exist"
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj:111]: why did we remove so many tests?

File: system-int-test/test/cmr/system_int_test/search/autocomplete_collection_facet_search_test.clj [hunk 6]
```diff
@@ -78,7 +78,7 @@
      (is (:CMR-Took headers))))
 
   (testing "entries return in descending score order"
-   (let [response (query->json-response-body "q=at")
+   (let [response (query->json-response-body "q=gulf")
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/autocomplete_collection_facet_search_test.clj:81]: why do we change this test?

File: system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj (hunk 5 above)
Comment (jaybarra) [reply system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj:111]: The existing tests didn't actually verify that a reindex happened successfully. The new test does

File: system-int-test/test/cmr/system_int_test/search/autocomplete_collection_facet_search_test.clj (hunk 6 above)
Comment (jaybarra) [reply system-int-test/test/cmr/system_int_test/search/autocomplete_collection_facet_search_test.clj]: changed by mistake, restored to previous.

File: system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj (hunk 5 above)
Comment (ygliuvt) [reply system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj:83]: There are tests on `organization` and `instrument` before. I think those are still supported by autocomplete. Why do you remove those?

Comment (jaybarra) [reply system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj:83]: I will add them back, the test was restructured to test for generic pass/fail and didn't look for other types.

**Pull request merged**